lib/sudoers/src/lib.rs

```diff
@@ -90,6 +90,15 @@ fn read_sudoers(path: &Path) -> Result<Vec<basic_parser::Parsed<Sudo>>, std::io:
     use std::io::Read;
     let mut source = File::open(path)?;
 
+    if !source.metadata()?.permissions().readonly() {
```

Comment:
wouldn't this reject a sudoers file with a chmod of `600`? the original sudo accepts such file. could you add another test case that uses that chmod string?

Reply:
by the way, original sudo will also work if the chmod string is "000". that is because root can read files that have no read permissions at all (same thing with write perms but not exec perms)

Reply:
Good point, I'll also check what it does with `464`.

Reply:
This is actually quite interesting and worthy of deserving more than this little 'quick fix', since if were going to permit write permissions in some cases, we need to check ownership too. ogsudo actually has a `secure_open` function for this that we may want to add to `sudo_system` soon too.

Reply:
After-dinner thought: the fact that reading the original C code doesn't immediately tell me what csudo is checking for opening the sudoers file, while a review of this code immediately tells us what we are doing, is maybe a very good indicator that a simpler solution is appropriate. If we discover a pattern we can extract the common functionality later.

Reply:
Solution with commit 713caa5 is now a more general solution (and should be conforming to csudo's behaviour). I added some ownership-testing-tests as well.